File: Daybreak.GWCA/source/payloads/SessionPayload.cpp
#include <cstdint>
#include <json.hpp>
#include <payloads/SessionPayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const SessionPayload& p) {
        j = json
        {
            {"FoedKilled", p.FoesKilled},
            {"FoesToKill", p.FoesToKill},
            {"MapId", p.MapId},
            {"InstanceTimer", p.InstanceTimer},
            {"InstanceType", p.InstanceType},
        };
    }
}

File: Daybreak/Services/Scanner/Models/SessionPayload.cs
namespace Daybreak.Services.Scanner.Models;
internal sealed class SessionPayload
{
    public uint FoesKilled { get; set; }
    public uint FoesToKill { get; set; }
    public uint MapId { get; set; }
    public uint InstanceTimer { get; set; }
    public uint InstanceType { get; set; }
}
